.github/workflows/build_3.3.0.yaml

```diff
@@ -20,15 +20,18 @@
 name: "Build and Test (3.3.0)"
 
 on:
-  pull_request:
+  pull_request_target:
     branches:
       - 'master'
     paths:
-      - '3.3.0/'
+      - '3.3.0/**'
+      - '.github/workflows/build_3.3.0.yaml'
       - '.github/workflows/main.yml'
 
 jobs:
   run-build:
+    permissions:
+      packages: write
```

Comment:
Is this needed ? This workflow just delegates to `main.yml` where this is also specified.
It does not do any harm either!

Reply:
It's needed see also: https://github.com/apache/spark/pull/38145